File: 1-Creational/Simple Factory/Java/src/PaymentTask/PaymentFactory.java
package PaymentTask;

public class PaymentFactory {

    public static Payment createPayment(String type) {
        if (type.equalsIgnoreCase("paypal")) {
            return new PaypalPayment();
        }

        if (type.equalsIgnoreCase("visa")) {
            return new VisaPayment();
        }

        if (type.equalsIgnoreCase("cash")) {
            return new CashPayment();
        }

        throw new IllegalArgumentException("Invalid payment type");
    }

}

File: CreationalDesignPatterns/PHP/PaymentSystem/PaymentFactory.php
<?php

namespace PaymentSystem;

class PaymentFactory
{
    public static function make($type)
    {
        switch ($type) {
            case 'paypal':
                return new PaypalPayment();

            case 'stripe':
                return new StripePayment();

            case 'cash':
                return new CashPayment();

            default:
                throw new Exception("Unsupported payment type");
        }
    }
}